### PR TITLE
[quantization] Changing observer name

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27779 [quantization] Changing observer name**
* #27670 [quantization] Refactoring names for consistency

Differential Revision: [D17886605](https://our.internmc.facebook.com/intern/diff/D17886605)

### CODE REVIEW

**Comment by zafartahirov** [comment]
@jerryzh168 I think I overdid that  :/

File: torch/nn/quantized/modules/conv.py [hunk 1]
```diff
@@ -222,20 +222,20 @@ def from_float(cls, mod):
                 mod.weight, mod.bias = \
                     fuse_conv_bn_weights(mod.weight, mod.bias, mod.running_mean,
                                          mod.running_var, mod.eps, mod.gamma, mod.beta)
-            assert hasattr(mod, 'observer'), 'Input QAT module must have observer attached'
+            assert hasattr(mod, 'activation_post_process'), 'Input QAT module must have observer attached'
             weight_observer = mod.weight_fake_quant
-            activation_observer = mod.observer
+            activation_observer = mod.activation_post_process
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py]: please change `activation_observer` to `activation_post_process` here as well

File: torch/nn/quantized/modules/conv.py [hunk 2]
```diff
@@ -222,20 +222,20 @@ def from_float(cls, mod):
                 mod.weight, mod.bias = \
                     fuse_conv_bn_weights(mod.weight, mod.bias, mod.running_mean,
                                          mod.running_var, mod.eps, mod.gamma, mod.beta)
-            assert hasattr(mod, 'observer'), 'Input QAT module must have observer attached'
+            assert hasattr(mod, 'activation_post_process'), 'Input QAT module must have observer attached'
             weight_observer = mod.weight_fake_quant
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py]: and `weight_observer` to `weight_post_process`?

**Comment by jerryzh168** [comment]
> @jerryzh168 I think I overdid that :/

what do you mean?

**Comment by z-a-f** [comment]
> > @jerryzh168 I think I overdid that :/
> 
> what do you mean?

I mean too many renames :)

**Review by jerryzh168 — APPROVED** [review]
LG

**Pull request closed without merging**